api/src/scheduler/data-export.ts

```diff
@@ -0,0 +1,111 @@
+import type { OpenAPIHono } from '@hono/zod-openapi';
+import type { EventLog } from '../routes/event-log/validation.ts';
+import type { Profile } from '../routes/profile/validation.ts';
+import type { Team } from '../routes/team/validation.ts';
+import type { PaginatedResponse } from '../utils/responses.ts';
+
+type TeamMember = Pick<Profile, 'avatar' | 'id' | 'name'>;
+
+async function getProfiles(app: OpenAPIHono<EnvironmentBindings>, env: Env, context: ExecutionContext) {
+	const results: Profile[] = [];
+
+	let lastJson: PaginatedResponse<Profile[]>;
+	do {
+		const response = await app.request('/api/profiles', { method: 'GET' }, env, context);
+		lastJson = await response.json() as PaginatedResponse<Profile[]>;
+
+		results.push(...lastJson.data);
+	} while (lastJson.currentPage !== lastJson.lastPage);
+
+	return results;
+}
+
+async function getTeams(app: OpenAPIHono<EnvironmentBindings>, env: Env, context: ExecutionContext) {
+	const results: Team[] = [];
+
+	let lastJson: PaginatedResponse<Team[]>;
+	do {
+		const response = await app.request('/api/teams', { method: 'GET' }, env, context);
+		lastJson = await response.json() as PaginatedResponse<Team[]>;
+
+		results.push(...lastJson.data);
+	} while (lastJson.currentPage !== lastJson.lastPage);
+
+	return results;
+}
+
+async function getTeamMembers(app: OpenAPIHono<EnvironmentBindings>, env: Env, context: ExecutionContext, teams: Team[]) {
+	const results: { teamId: string, members: TeamMember[] }[] = [];
+
+	for (const team of teams) {
+		const teamMembers: TeamMember[] = [];
+
+		let lastJson: PaginatedResponse<TeamMember[]>;
+		do {
+			const response = await app.request(`/api/teams/${team.id}/members`, { method: 'GET' }, env, context);
+			lastJson = await response.json() as PaginatedResponse<TeamMember[]>;
+
+			teamMembers.push(...lastJson.data);
+		} while (lastJson.currentPage !== lastJson.lastPage);
+
+		results.push({
+			teamId: team.id,
+			members: teamMembers
+		});
+	}
+
+	return results;
+}
+
+async function getEventLog(_app: OpenAPIHono<EnvironmentBindings>, _env: Env, _context: ExecutionContext) {
+	// TODO: implement when we have endpoints for event logs
```

Comment:
Nit pick: Should we get in the habit of creating issues whenever we utilize TODO comments and add the linke below them?